Compiler warning in soc/xtensa/esp32/include/_soc_inthandlers.h

Commit: 7cea94942c35bb65a8c219b69ec2270c0fedd6f4

Log:

```
In file included from /home/kspoorth/work/forked_zephyr/arch/xtensa/core/xtensa-asm2.c:14:0:
/home/kspoorth/work/forked_zephyr/soc/xtensa/esp32/include/_soc_inthandlers.h: In function '_xtensa_handle_one_int0':
/home/kspoorth/work/forked_zephyr/soc/xtensa/esp32/include/_soc_inthandlers.h:363:1: warning: label 'handle_irq' defined but not used [-Wunused-label]
 handle_irq:
 ^
/home/kspoorth/work/forked_zephyr/soc/xtensa/esp32/include/_soc_inthandlers.h: In function '_xtensa_handle_one_int6':
/home/kspoorth/work/forked_zephyr/soc/xtensa/esp32/include/_soc_inthandlers.h:373:1: warning: label 'handle_irq' defined but not used [-Wunused-label]
 handle_irq:
 ^
```

It is good to get rid of this as it would be an issue while building with compiler warnings being treated as errors.

--- a/arch/xtensa/core/xtensa_intgen.py
+++ b/arch/xtensa/core/xtensa_intgen.py
@@ -123,11 +123,15 @@ for lvl in ints_by_lvl:
     cprint("static inline int _xtensa_handle_one_int" + str(lvl) + "(unsigned int mask)")
     cprint("{")
 
+    if not ints_by_lvl[lvl]:
+        cprint("return 0;")
+        cprint("}")
+        continue
+
     cprint("int irq;")
     print("")
 
-    ints_by_lvl[lvl].sort()
-    emit_int_handler(ints_by_lvl[lvl])
+    emit_int_handler(sorted(ints_by_lvl[lvl]))
 
     cprint("return 0;")
     cprint("handle_irq:")

--- a/soc/xtensa/esp32/include/_soc_inthandlers.h
+++ b/soc/xtensa/esp32/include/_soc_inthandlers.h
@@ -357,21 +357,9 @@ handle_irq:
 
 static inline int _xtensa_handle_one_int0(unsigned int mask)
 {
-	int irq;
-
 	return 0;
-handle_irq:
-	_sw_isr_table[irq].isr(_sw_isr_table[irq].arg);
-	return mask;
 }
-
 static inline int _xtensa_handle_one_int6(unsigned int mask)
 {
-	int irq;
-
 	return 0;
-handle_irq:
-	_sw_isr_table[irq].isr(_sw_isr_table[irq].arg);
-	return mask;
 }
-

